static/app/main.tsx

```diff
@@ -29,7 +30,9 @@ function Main() {
         <ThemeAndStyleProvider>
           <NuqsAdapter defaultOptions={{shallow: false}}>
             <CommandPaletteProvider>
-              <RouterProvider router={router} />
+              <RouteConfigProvider value={router.routes}>
+                <RouterProvider router={router} />
+              </RouteConfigProvider>
```

Comment:
note: react-router does not expose a hook that gives us access to all the routes; the recommended way is to just export your routs (“it’s just config”), but that ties it to a specific implementation and makes it hard to test.

instead, I’ve opted for another provider that allows access to the route config of the router that gets passed to `RouterProvider`.

This is the _only_ place where we add this. `linkBehavior` reads the context to enable intent prefetching, and if there is no context, we simply don’t prefetch anything. That means we don’t need to add this provider everywhere.